### PR TITLE
Add build pipeline ReleaseType=Weekly

### PR DESCRIPTION
**Pull request by andrew-m-leonard**
Add the "Weekly" ReleaseType choice to the build pipeline job, which does the equivalent of a Nightly but with the Weekly test jobs, and then publishes to github.

Signed-off-by: Andrew Leonard <anleonar@redhat.com>

### CODE REVIEW

**Comment by andrew-m-leonard** [comment]
run tests

**Comment by andrew-m-leonard** [comment]
run tests

**Review by lumpfish — APPROVED** [review]
Looks good

**Review by sxa — COMMENTED** [review]
Is this what we will be using for the OpenJ9 milestone builds? If so that should be called out in the RELEASING.md doc

**Comment by andrew-m-leonard** [comment]
> Is this what we will be using for the OpenJ9 milestone builds? If so that should be called out in the RELEASING.md doc

Updated RELEASING.md

**Comment by andrew-m-leonard** [comment]
run tests

File: pipelines/build/common/build_base_file.groovy [hunk 1]
```diff
@@ -727,6 +731,7 @@ return {
             enableSigner: Boolean.parseBoolean(enableSigner),
             publish: publish,
             release: release,
+            buildTrigger: releaseType,
```
Comment (sxa) [inline pipelines/build/common/build_base_file.groovy:734]: What's the thinking behind have `buildTrigger` and `releaseType` - two differnet names referring to the same thing. That's likely to cause confusion

**Review by sxa — COMMENTED** [review]
Another comment on the naming convention

File: pipelines/build/common/build_base_file.groovy (hunk 1 above)
Comment (andrew-m-leonard) [reply pipelines/build/common/build_base_file.groovy:734]: yeah true, I was using the "terminology" already being used in the setTestList() method which mentioned "build trigger".... and releaseType is the existing variable being passed in "Nightly", "Nightly with no publish", "Release"
i'll change to releaseType

Comment (andrew-m-leonard) [reply pipelines/build/common/build_base_file.groovy]: changed

**Comment by andrew-m-leonard** [comment]
run tests

**Review by M-Davies — APPROVED** [review]

**Review by sxa — APPROVED** [review]
Thanks for the naming fix :-)

**Pull request merged**